romancal/outlier_detection/tests/test_outlier_detection.py

```diff
@@ -1,42 +1,358 @@
+        the telescope reference system onto the world reference systems.
+    """
+    v2_ref = input_dm.meta.wcsinfo.v2_ref / 3600
+    v3_ref = input_dm.meta.wcsinfo.v3_ref / 3600
+    roll_ref = input_dm.meta.wcsinfo.roll_ref
+    ra_ref = input_dm.meta.wcsinfo.ra_ref
+    dec_ref = input_dm.meta.wcsinfo.dec_ref
+
+    angles = np.array([v2_ref, -v3_ref, roll_ref, dec_ref, -ra_ref])
+    axes = "zyxyz"
+    rot = RotationSequence3D(angles, axes_order=axes)
+
+    # The sky rotation expects values in deg.
+    # This should be removed when models work with quantities.
+    model = (
+        (Scale(0.1 / 3600) & Scale(0.1 / 3600))
+        | SphericalToCartesian(wrap_lon_at=180)
+        | rot
+        | CartesianToSpherical(wrap_lon_at=360)
+    )
+    model.name = "v23tosky"
+    return model
+
+
+def _create_wcs(input_dm, shift_1=0, shift_2=0):
+    """
+    Create a basic WCS object (with optional shift) and
+    append it to the input_dm.meta attribute.
+
+    The WCS object will have a pipeline with the required
+    steps to validate against the TweakReg pipeline.
+
+    Parameters
+    ----------
+    input_dm : roman_datamodels.ImageModel
+        A Roman image datamodel.
+    shift_1 : int, optional
+        Shift to be applied in the direction of the first axis, by default 0
+    shift_2 : int, optional
+        Shift to be applied in the direction of the second axis, by default 0
+    """
+
+    shape = input_dm.data.shape
+
+    # create necessary transformations
+    distortion = Shift(-shift_1) & Shift(-shift_2)
+    distortion.bounding_box = ((-0.5, shape[-1] + 0.5), (-0.5, shape[-2] + 0.5))
+    tel2sky = _create_tel2sky_model(input_dm)
+
+    # create required frames
+    detector = cf.Frame2D(name="detector", axes_order=(0, 1), unit=(u.pix, u.pix))
+    v2v3 = cf.Frame2D(
+        name="v2v3",
+        axes_order=(0, 1),
+        axes_names=("v2", "v3"),
+        unit=(u.arcsec, u.arcsec),
+    )
+    world = cf.CelestialFrame(reference_frame=coord.ICRS(), name="world")
+
+    # create pipeline
+    pipeline = [
+        wcs.Step(detector, distortion),
+        wcs.Step(v2v3, tel2sky),
+        wcs.Step(world, None),
+    ]
+
+    wcs_obj = wcs.WCS(pipeline)
+
+    input_dm.meta["wcs"] = wcs_obj
+
+
+@pytest.fixture
+def base_image():
+    """
+    Create a base image with a realistic WCSInfo and a WCS.
+
+    Notes
+    -----
+    The size of the image needs to be relatively large in order for
+    the source catalog step to find a reasonable number of sources in the image.
+
+    shift_1 and shift_2 (units in pixel) are used to shift the WCS projection plane.
+    """
+
+    def _base_image(shift_1=0, shift_2=0):
+        l2 = maker_utils.mk_level2_image(shape=(100, 100))
+        l2_im = rdm.ImageModel(l2)
+        _create_wcs(l2_im)
+        l2_im.meta.wcsinfo.vparity = -1
+        return l2_im
+
+    return _base_image
+
+
+def create_asn_file(tmp_path, members_mapping=None):
```

Comment:
This looks like another [code duplicate](https://github.com/spacetelescope/romancal/blob/main/romancal/tweakreg/tests/test_tweakreg.py#L88). Make it a fixture ?